.github/workflows/workflow.yml

```diff
@@ -215,3 +217,32 @@ jobs:
           name: sphinx-docs
           path: docs/build
           retention-days: 7
+
+  osx_arm:
```

Comment:
Why is this added in the current PR?

Reply:
Hey, sure, probably I should have just renamed the PR at some point... I wanted to add better Apple coverage since that was the genesis of this PR. More details below.

This PR accomplishes two things, improving OSX M1 support and swapping the FFT.  Swapping the default FFT away from `pyfftw` was required to support OSX on M1.  Since that was the main code change that is what ended up being the branch and pr name.

Previously the code was not able to run using the ARM M1 native packages (only in x86_64 emulation which was the conda default until relatively recently). According to github this osx 14 image is aarch64 (arm), and is supposed to be equivalent to M1.  I added this job to provide better coverage.  Otherwise the only way we can test this is on people's personal machines manually...  We have noticed that not every apple silicon machine is performing identically, but its better than nothing...

Reply:
(Well I should be careful.  Our code was able to run, but required building and installing multiple packages from source on an M1 for running natively.  It was broken via pip).